Sources/FoundationEssentials/Calendar/Calendar_Recurrence.swift

```diff
@@ -0,0 +1,846 @@
+        lazy var monthRange = calendar.range(of: .month, in: .year, for: anchor)!
+        let months = months.map { month in
+            if month.index > 0 {
+                return month
+            } else {
+                let newIndex = monthRange.upperBound + month.index
+                // The upper bound is the last month plus one. Subtracting 1 we get the last month
+                return Calendar.RecurrenceRule.Month(newIndex, isLeap: month.isLeap)
+            }
+        }
+        
+        if action == .limit {
+            dates = dates.filter {
+                let idx = calendar.component(.month, from: $0)
+                let isLeap = calendar._dateComponents([.month], from: $0).isLeapMonth
+                return months.contains {
+                    $0.index == idx && $0.isLeap == isLeap
+                }
+            }
+        } else {
+            let componentSet: Calendar.ComponentSet = [ .month, .isLeapMonth, .day, .hour, .minute, .second, .nanosecond ]
+            
+            let anchorComponents = calendar._dateComponents(componentSet, from: anchor)
+            let daysInYear = calendar.dateInterval(of: .year, for: anchor)!
+            // This is always the first expansion, so we can overwrite `dates`
+            dates = months.compactMap { month in
+                var components = anchorComponents
+                components.month = month.index
+                components.isLeapMonth = month.isLeap
+                return calendar.nextDate(after: daysInYear.start, matching: components, matchingPolicy: matchingPolicy)
+            }
+        }
+    }
+    internal func _expandOrLimitDaysOfTheMonth(dates: inout [Date], anchor: Date, action: ComponentAction) {
+        if action == .limit {
+            dates = dates.filter { date in
+                let day = calendar.component(.day, from: date)
+                let dayRange = calendar.range(of: .day, in: .month, for: date)
+                return daysOfTheMonth.map {
+                    if $0 > 0 {
+                        $0
+                    } else {
+                        dayRange!.upperBound + $0
+                    }
+                }.contains(day)
+            }
+        } else {
+            let components: Calendar.ComponentSet = [
+                .day,
+                .hour,
+                .minute,
+                .second,
+                .nanosecond
+            ]
+            let anchorComponents = calendar._dateComponents(components, from: anchor)
+            
+            var componentsForEnumerating: [DateComponents] = []
+            let enumerationDateInterval = calendar.dateInterval(of: frequency.component, for: anchor)!
+            
+            if frequency == .yearly {
+                let monthRange = calendar.range(of: .month, in: .year, for: anchor)!
+                let firstDayOfYear = enumerationDateInterval.start
+                let monthsToDaysInMonth = monthRange.reduce(into: [Int: Int]()) {
```

Comment:
We don't need this if every `day` in `daysOfTheMonth` is greater than 0 (if we enter the `if day > 0 {` closure below at line 552). 
Can we calculate this more lazily?